### PR TITLE
chore: use fork of skim

### PR DESCRIPTION
**Pull request by conradludgate**
We've seen that adding skim broke our tentative windows-support, and I did have some quarrels with the skim code. 

As suggested by kimono-koans, we can use fuzzy-matcher as a lib only impl of skim

### CODE REVIEW

**Comment by kimono-koans** [comment]
FYI about my fork/branch -- "vendored" is the branch I use for [httm](https://github.com/kimono-koans/httm) and is for all intents and purposes my *production* branch, but is not supported by me except for use with `httm`.  

I think I've made PRs of all my changes that don't include API changes.  They are:

https://github.com/lotabout/skim/pull/507
https://github.com/lotabout/skim/pull/505
https://github.com/lotabout/skim/pull/503

https://github.com/lotabout/skim/pull/505 is probably something you will want to incorporate, as it makes the UI much more responsive during searches of more than 100_000 items.

The only other major changes that comes to mind is [ingest](https://github.com/kimono-koans/two_percent/blob/vendored/src/helper/ingest.rs), removal of [explicit static lifetimes](https://github.com/kimono-koans/two_percent/blob/254d5775c920ae5fc2ef833faaf62b8cb3da5d7d/src/helper/item.rs#L20), and attempts to reduce the overhead of common structs (see last link as well).  These all would seem to require API changes and therefore may or may not be of interest to you.

Re: implicit static lifetimes with trait objects, I am sure you are all more clever than me, so I hope one of you can find a fix.  I've tried HRTB, which is "trait_objects_2" and which don't actually work, lifetimes with PhantomData (which I never could get to compile), and custom Drop implementations (which still seem most promising?), and, although now all my ingested structs are deallocated, my MatchedItems still appear not to be.

Good luck!

**Comment by YummyOreo** [comment]
> Hopefully we can contribute a patch to make all of their tui dependencies optional

I can try to make a patch w/ a feature flag to disable all tui stuffs later today

File: src/command/client/search/engines/skim.rs [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+pub struct Search {
+    all_history: Vec<(History, i32)>,
+    engine: SkimMatcherV2
+}
+
+impl Search {
+    pub fn new() -> Self {
+        Search {
+            all_history: vec![],
+            engine: SkimMatcherV2::default(),
```
[the remark this answers is not in the document]
Comment (ellie) [reply src/command/client/search/engines/skim.rs:21]: Not toooo concerned about full parity - though we should probably limit history size client side too

[the remark this answers is not in the document]
Comment (YummyOreo) [reply src/command/client/search/engines/skim.rs:21]: > though we should probably limit history size client side too

Is this for each history entry or for the whole history?

**Review by ellie — APPROVED** [review]
LGTM

**Comment by conradludgate** [comment]
Oh, that rebase broke more things 😩

File: src/command/client/search/engines/skim.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (ellie) [reply src/command/client/search/engines/skim.rs:21]: Each history entry! It's not supposed to store shell-script sized history lines, and I'd like to do "large file" type support for that use case. Not just a huge row

**Review by ellie — APPROVED** [review]

**Pull request merged**